pkg/diagnostics/network/run_pod.go

```diff
@@ -0,0 +1,167 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/diagnostics/networkpod/util"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+const (
+	NetworkDiagnosticName = "NetworkCheck"
+)
+
+// NetworkDiagnostic is a diagnostic that runs a network diagnostic pod and relays the results.
+type NetworkDiagnostic struct {
+	KubeClient          *kclient.Client
+	OSClient            *osclient.Client
+	ClientFlags         *flag.FlagSet
+	Level               int
+	Factory             *osclientcmd.Factory
+	PreventModification bool
+	LogDir              string
+
+	pluginName   string
+	nodes        []kapi.Node
+	nsName       string
+	globalnsName string
+	res          types.DiagnosticResult
+}
+
+// Name is part of the Diagnostic interface and just returns name.
+func (d *NetworkDiagnostic) Name() string {
+	return NetworkDiagnosticName
+}
+
+// Description is part of the Diagnostic interface and provides a user-focused description of what the diagnostic does.
+func (d *NetworkDiagnostic) Description() string {
+	return "Create a pod on all schedulable nodes and run network diagnostics from the application standpoint"
+}
+
+// CanRun is part of the Diagnostic interface; it determines if the conditions are right to run this diagnostic.
+func (d *NetworkDiagnostic) CanRun() (bool, error) {
+	if d.PreventModification {
+		return false, errors.New("running the network diagnostic pod is an API change, which is prevented as you indicated")
+	} else if d.KubeClient == nil {
+		return false, errors.New("must have kube client")
+	} else if d.OSClient == nil {
+		return false, errors.New("must have openshift client")
+	} else if _, err := d.getKubeConfig(); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// Check is part of the Diagnostic interface; it runs the actual diagnostic logic
+func (d *NetworkDiagnostic) Check() types.DiagnosticResult {
+	d.res = types.NewDiagnosticResult(NetworkDiagnosticName)
+
+	var err error
+	var ok bool
+	d.pluginName, ok, err = util.GetOpenShiftNetworkPlugin(d.OSClient)
+	if err != nil {
+		d.res.Error("DNet2001", err, fmt.Sprintf("Checking network plugin failed. Error: %s", err))
+		return d.res
+	}
+	if !ok {
+		d.res.Warn("DNet2002", nil, fmt.Sprintf("Skipping network diagnostics check. Reason: Not using openshift network plugin."))
```

Comment:
Likewise here, I'd like to see this in CanRun() not as a warning. That way running it in a devenv doesn't get you a useless warning.